reana_commons/validation/schemas/reana_analysis_schema.json

```diff
@@ -159,6 +159,26 @@
         }
       }
     },
+    "tests": {
+      "id": "/properties/tests",
+      "type": "object",
+      "title": "Analysis tests.",
+      "additionalProperties": false,
+      "properties": {
+        "files": {
+          "id": "/properties/tests/properties/files",
+          "type": "array",
+          "title": "Analysis test files.",
+          "description": "Gherkin test feature files.",
+          "optional": true,
```

Comment:
The `optional` keyword was used in JSONSchema v2 ([draft 2 specification](https://datatracker.ietf.org/doc/html/draft-zyp-json-schema-02)), but is not present in v4 ([draft 4 specification](https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00)). The [changelog of v3](https://datatracker.ietf.org/doc/html/draft-zyp-json-schema-03#anchor54) shows how they `Replaced "optional" attribute with "required" attribute`. Since the specified JSONSchema for the reana.yaml schema is draft 4 (at the beginning of this file), maybe we can remove the line with `"optional"`, even if it does no harm.
We can maybe also remove it from the rest of the file, since it may be misleading...

What do you think?

Reply:
Nice catch! Indeed after reading more about it it seems that all properties are optional by default now in v4 spec.. I've removed all the `optional` fields and tested it locally, seems to be working just fine